docs/Contributing.md

```diff
@@ -0,0 +1,84 @@
+
+## Outline
+
+As you begin adding code to the repository you will notice there is a lot of code. And we have a standard naming convention for folders and files that you should follow.
+
+1. Fork the repository, and keep your `master` branch the same as `messari/subgraphs`, before creating a feature branch in your repository **pull** upstream changes.
+2. Create your feature branch `branch-name`.
+3. Once you begin to **push** changes create a Pull Request (PR) in `messari/subgraphs` and set it as a draft until you it is ready for review.
+
+![How to convert to draft](./images/contributing/convert-to-draft.png)
+
+> See "Naming Conventions" for how to name your PR
+
+4. Once your PR is ready for review go ahead and click "Ready for review". As a courtesy it is nice to let the reviewer know that it is ready through a DM.
+
+At this point you will just need to follow the "Reviewing" process until your PR is merged.
+
+## Reviewing
+
+### Reviewee
+
+- Generally you will need to make changes, either respond to a change/comment with a question, rebuttle, or comment. Otherwise, make the change and resolve the issue.
+- When you feel changes are sufficient, let the reviewer know again that the PR is ready for another round of reviews.
+- Do not make a new PR for changes, this will make it harder for the reviewer to track their progress.
+
+### Reviewer
+
+- This part is straight forward, and if you are reviewing you know what to do.
+- It is courtesy to let the developer know that you are done with the review.
+
+> This is an iterative process that takes time, so don't think it will always be easy.
+
+## Rebasing
+
+There are multiple ways to do this. Most of the time you will rebase onto `master`, so you should make sure your `master` branch is up to date with the upstream `master` (ie, Messari's repo).
+
+Go into your feature-branch and make the following call.
+
+```bash
+git pull --rebase origin master
+```
+
+This will initiate a rebase. Sometimes you will have to resolve conflicts. And then `git add .` and `git rebase --continue` or `git rebase --skip`.
+
+Following this method, after rebasing you can do the command `git push --force` to update your remote repo.
+
+## Naming Convention
+
+### Pull Requests
+
+It is nice to have a consistent naming convention for pull requests. Oftentimes there are dozens of PRs out on `messari/subgraphs` so being able to know exactly what a PR is is important.
+
+PR names also drive the commit name once a PR is merged into `master`. In this way it is easier to tell what was changed in each commit.
+
+An impact-level identifier:
+
+### 1️⃣
+
+- Subgraphs: change impacts an entire fork directory / multiple protocols
+- Dashboard: change impacts the entire dashboard / multiple objects
+
+### 2️⃣
+
+- Subgraphs: change impacts a single protocol
+- Dashboard: change impacts a single object
+
+### 3️⃣
```

Comment:
I like the idea of the impact level identifier. How do you feel about making them more descriptive? 

For example, using semver wording instead of numbers? (#patch, #minor, #major)
Or something along the lines?

Imagine a change which has no impact on code logic but affects many files from different protocols? Like formatting all files after adding some linter o something? Would it be a `1` or a `3`? It could be a 1 because it affects a lage portion of the repo, but also a 3 because no functionality was changed. I feel that by using words we can make it more explicit.

Reply:
@jaimehgb what do you think about keeping the numbers but the meanings correspond to semver wording. So 1 = #major 2 = #minor and 3 = #patch ? The idea would be to keep the PR names shorter

Reply:
or we could use words like `feat` `fix` `docs` seen here https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary

Reply:
If the concern is PR name lenght we can use github labels instead 😅 

Apart from giving it a clearer meaning, part of the reason I personally prefer words over numbers is to avoid having to mentally map:
>2 -> oh, this was minor, right?
>or was it?
It might sound silly but 🤷‍♂️ 

And if we start adding other levels, like they mention about `other` in the link you share it might get messy.

I'm not so strongly opinionated on the wording we use though. Using conventionalcommits' schema sounds good to me too :D

Reply:
@jaimehgb i didn't go with the comments I made lol. lmk your thoughts on the current change though

Reply:
lmao

It looks good to me, pretty descriptive 🚀 
Do you want to keep semver together with the rest of names? Since it can be inferred from each

Reply:
@jaimehgb what do you mean by keep it together? Sorry I don't quite understand what you mean

Reply:
Oh, re-reading myself I didn't explain very well.

I was asking if you wanted to keep both semver (major, minor, patch) and the rest of identifiers (fix, feat, docs, etc...) for PR naming. Since you can infer semver from the others it might be redundant:
- fix -> patch
- feat -> minor
- docs -> _none_
- ...

Althugh now that I think about it, thinking out loud, some of them might not be so direct (chore, style, perf). So disregard my question 😅

Reply:
okay lol, I like the descriptiveness of the schema i came up with but I don't like the aesthetic, but I think I can look past that lol